### ISSUE TITLE
CTC loss doubt

### ISSUE THREAD

**Issue by Dreyer1416** [SeanNaren/deepspeech.pytorch#194]
Does the CTC loss function calculates the softmax activation function internally?

**Comment by SeanNaren** [comment]
Yes, which is why we have a separate soft max that only works in inference, not in training!

**Comment by Dreyer1416** [comment]
Ok. Thanks for the answer!

 I noticed the separated softmax in inference, but I was still confused because in this [test file](https://github.com/SeanNaren/warp-ctc/blob/pytorch_bindings/pytorch_binding/tests/test.py) the CTC appears to be receiving the output of a softmax function. 

Maybe it would be I good idea to update the test function to avoid further misinterpretation
